package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua

```diff
@@ -1,7 +1,66 @@
+local uci = require("simple-uci").cursor()
+
+local util = require "gluon.util"
+
 local M = {}
 
-function M.get_mesh_vpn_interface()
-  return 'mesh-vpn'
+function M.enabled()
+	return uci:get_bool("gluon", "mesh_vpn", "enabled")
+end
+
+function M.enable(val)
+	return uci:set("gluon", "mesh_vpn", "enabled", val)
+end
+
+function M.get_interface()
+	return 'mesh-vpn'
+end
+
+function M.get_method(name)
+	return require("gluon.mesh-vpn.method." .. name)
+end
+
+function M.get_method_names()
+	local out = {}
+
+	for _, v in ipairs(util.glob("/lib/gluon/mesh-vpn/method/*")) do
+		table.insert(out, v:match('([^/]+)$'))
+	end
+
+	return out
+end
+
+function M.get_active_method()
+	local name = nil
+	local method = nil
+
+	for _, n in ipairs(M.get_method_names()) do
+		method = M.get_method(n)
+
+		if method.active() then
+			name = n
+		end
+	end
+
+	if method ~= nil then
+		return name, method
+	end
+
+	return nil, nil
+end
+
+function M.get_inactive_methods()
+	local methods = {}
+
+	for _, name in ipairs(M.get_method_names()) do
+		local method = M.get_method(name)
+
+		if not method.active() then
```

Comment:
Related to https://github.com/freifunk-gluon/gluon/pull/2102/files#r626032365
```suggestion
		if not method.available() then
```